libp2p/muxers/yamux/yamux.nim

```diff
@@ -279,10 +279,15 @@ method readOnce*(
     raise newLPStreamRemoteClosedError()
   if channel.recvQueue.len == 0:
     channel.receivedData.clear()
-    try: # https://github.com/status-im/nim-chronos/issues/516
-      discard await race(channel.closedRemotely.wait(), channel.receivedData.wait())
-    except ValueError:
-      raiseAssert("Futures list is not empty")
+    let
+      closedRemotelyFut = channel.closedRemotely.wait()
+      receivedDataFut = channel.receivedData.wait()
+    defer:
+      if not closedRemotelyFut.finished():
+        await closedRemotelyFut.cancelAndWait()
+      if not receivedDataFut.finished():
+        await receivedDataFut.cancelAndWait()
+    await closedRemotelyFut or receivedDataFut
```

Comment:
It is probably better to use
```nim
proc raceCancel*(
    futs: seq[InternalRaisesFuture]
): Future[FutureBase] {.async: (raises: [ValueError, CancelledError]).} =
  try:
    race(futs)
  finally:
    await noCancel allFutures(futs.filterIt(not it.finished).mapIt(it.cancelAndWait))
```

Reply:
It's also going to handle cancelations.

Reply:
I'm not a huge fan of using `InternalRaisesFuture` outside of Chronos tbh.

Reply:
Me neither, but that's the only way for now, unless SomeFuture is changed to public.